### ISSUE TITLE
subcube task is slow in deciding if cube is outside range

### ISSUE THREAD

**Issue by apopping** [SoFiA-Admin/SoFiA#73]
It takes quite a while for SoFiA to decide whether a given subcube is within the ranges of the input cube.
This can be problematic when the optical finder is being used with a large input catalogue.

I have tested this on my laptop using a 10Gb input cube. It takes about ~8 seconds do determine whether a  source in the catalogue is outside the cube and move on to the next source.

In case of the  optical finder we could load the header of the input cube to get all the coordinate information and do this check for every catalogue entry before actually starting the sofia pipeline, but maybe there is better solution to speed this up in general.

**Comment by paoloserra** [comment]
What is taking most of this time? Is it "hdulist = fits.open(inFile)" (line 29)?

**Comment by apopping** [comment]
I haven't looked into this in detail, but I suspect this is the only place where things can really be slowed down. The rest are simple "if" checks that should run really fast.

I think this only really is an issue for the optical finder where in the current mode the header information is being imported for every single catalogue entry  (independent whether the subcube is valid or not).
It would be much better to do this only once and then only extract the actual data values for every catalogue entry

**Comment by ngiese** [comment]
I think, it is a good idea to check first for all subcubes if they lie within the boundaries and then start the pipeline. One thing that might slow things down a bit is the loading of the different python modules. Maybe a good idea is to put some more time outputs in the pipeline to see what is taking up how much time.

**Comment by paoloserra** [comment]
Yes I agree that we should first check, and if the problem is fits.open() we should modify the code. We may have to write a separate import module for the optical finder.
